calico/openstack/mech_calico.py

```diff
@@ -100,26 +103,52 @@ def __init__(self):
         # initialize these properly when we first need them.
         self.db = None
         self.transport = None
-        self._state_initialized = False
+        self._my_pid = None
+        self._periodic_resync_greenlet = None
+        self._epoch = 0
 
-        # Start our resynchronization process.
-        eventlet.spawn(self.periodic_resync_thread)
+        # Make sure we initialise even if we don't see any API calls.
+        eventlet.spawn_after(STARTUP_DELAY_SECS, self._init_state)
 
     def _init_state(self):
         """
         Creates the connection state required for talking to the Neutron DB
         and to etcd. This is a no-op if it has been executed before.
         """
-        if self._state_initialized:
+        current_pid = os.getpid()
+        if self._my_pid == current_pid:
+            # We've initialised our PID and it hasn't changed since last time,
+            # nothing to do.
```

Comment:
Yeah, the delta over field access isn't bad, but that wasn't really my question: my question was whether we feel the need to do the PID check every time. I guess I can see the value, though we should remember that we're costing ourselves CPU time each time we do it. Yeah, the database is way more expensive, but we have far less control over the DB than we do over our own code.